service/src/server/routes/deployment.rs

```diff
@@ -0,0 +1,149 @@
+/// free query token to graph node
+/// Later add receipt manager functions for paid queries
+pub async fn deployment_health(
+    Extension(server): Extension<ServerOptions>,
+    deployment: axum::extract::Path<String>,
+) -> impl IntoResponse {
+    // Create the GraphQL query
+    let query = status_query(deployment.to_string());
+
+    // Send the GraphQL request
+    let response = reqwest::Client::new()
+        .post(server.graph_node_status_endpoint)
+        .header("Content-Type", "application/json")
+        .json(&query)
+        .send()
+        .await;
+
+    match response {
+        Ok(response) => {
+            if response.status().is_success() {
+                // Deserialize the JSON response
+                //TODO: match with error
+                let data: serde_json::Value = if let Ok(data) = response.json().await {
+                    data
+                } else {
+                    return internal_server_error_response("Invalid json response");
+                };
+
+                // Process the response and return the appropriate HTTP status
+                let status = if let Some(status) =
+                    data["data"]["indexingStatuses"].get(0).and_then(|s| {
+                        let parse = serde_json::from_value::<IndexingStatus>(s.clone());
+                        parse.ok()
+                    }) {
+                    status
+                } else {
+                    return internal_server_error_response("Missing indexing status");
+                };
+
+                // Build health response based on the returned status
+                if status.health == "failed" {
+                    return internal_server_error_response("Subgraph deployment has failed");
+                }
+
+                if let Ok((latest, head)) = block_numbers(status) {
+                    if latest > head - 5 {
+                        (
+                            StatusCode::OK,
+                            axum::response::AppendHeaders([(
+                                HeaderName::from_static("graph-attestable"),
+                                "true",
+                            )]),
+                            Json("Subgraph deployment is up to date"),
+                        )
+                            .into_response()
+                    } else {
+                        internal_server_error_response("Subgraph deployment is lagging behind")
+                    }
+                } else {
+                    internal_server_error_response(
+                        "Invalid indexing status (missing block numbers)",
+                    )
+                }
+            } else {
+                internal_server_error_response("Unknown error")
+            }
+        }
+        Err(e) => internal_server_error_response(&e.to_string()),
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct IndexingStatus {
+    health: String,
```

Comment:
Could it make sense to go the extra mile and use an Enum here?

```rs
#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(non_camel_case_types)] // Need exact field names to match with GQL response
enum SubgraphHealth {
    healthy,
    unhealthy,
    failed
}
```

Reply:
another idea is maybe we could use [serde alias](https://serde.rs/field-attrs.html), but I'm fine with the current way:)